src/main/java/thePackmaster/patches/RenderBaseGameCardPackTopTextPatches.java

```diff
@@ -49,4 +54,25 @@ public static void patch(AbstractCard __instance, SpriteBatch sb) {
             }
         }
     }
+
+    public static boolean shouldShowPackName(AbstractCard c) {
+        return !Settings.hideCards
+                && c.getClass().getSuperclass().equals(AbstractCard.class)
+                && (isInPackmasterRun() || isInPackmasterCardLibraryScreen());
+    }
+
+    private static boolean isInPackmasterRun() {
+        return AbstractDungeon.player != null && AbstractDungeon.player.chosenClass == ThePackmaster.Enums.THE_PACKMASTER;
+    }
+
+    private static boolean isInPackmasterCardLibraryScreen() {
```

Comment:
This function is the new code -- the other two functions above are just me cleaning up and reorganizing the previously existing logic so that it was easier to follow.

The logic here isn't too bad, but it all piggy-backs on the BaseMod ColorTabBarFix patches, so to understand it you really need to understand what BaseMod is doing.